controllers/machineinventory_controller.go

```diff
@@ -329,20 +329,27 @@ func (r *MachineInventoryReconciler) ignoreIncrementalStatusUpdate() predicate.F
 		// Avoid reconciling if the event triggering the reconciliation is related to incremental status updates
 		// for MachineInventory resources only
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			if e.ObjectOld.GetObjectKind().GroupVersionKind().Kind != "MachineInventory" {
-				return true
```

Comment:
My manual tests revealed we were always hitting this path. `Kind` is always returning an empty string. Casting the object seams to be the way to go.